src/runtime/objmodel.cpp

```diff
@@ -458,38 +463,45 @@ BoxedHeapClass::BoxedHeapClass(BoxedClass* base, gcvisit_func gc_visit, int attr
     memset(&as_buffer, 0, sizeof(as_buffer));
 }
```

Comment:
Since the other constructors use BoxedString, I have to dance around to get it to work for any types initialized before `str_cls` and then initialize the names later. Maybe there's a better way but nothing jumped out at me.

I think I should probably do something to make this constructor harder to call on accident, since I only intend it for that purpose.